src/framework/operations.hpp

```diff
@@ -369,6 +373,17 @@ inline Op make_mat(const reg_t &qubits, const cmatrix_t &mat, std::string label
   return op;
 }
 
+inline Op make_matrixes(const std::vector<reg_t> &qubitss, const std::vector<cmatrix_t> &mats, std::string label = "") {
```

Comment:
`make_matrixes->make_matrix_sequence`

Here you could loop over the `qubit_seq` arguments and sort them and store the sorted qubits as `op.qubits` rather than do it later. 

This would also make it easier to combine two `matrix_sequence` instructions as you already have the sorted list for each and could just convert them to a set take the union, and convert back to a vector for example.

Reply:
In plan, there are some optimization based on a matrix in matrix_sequence.  Refactor around make_matrix_sequence when these optimization is implemented.